sycl/doc/extensions/supported/sycl_ext_oneapi_backend_level_zero.md

```diff
@@ -349,6 +421,15 @@ Applications must make sure that the Level-Zero handles themselves aren't used s
 Practically speaking, and taking into account that SYCL runtime takes ownership of the Level-Zero handles,
 the application should not attempt further direct use of those handles.
 
+### 4.5 Interoperability buffer synchronization rules
+
+A SYCL buffer that is constructed with this interop API uses the Level Zero memory allocation for its full lifetime, and the contents of the Level Zero memory allocation are unspecified for the lifetime of the SYCL buffer. If the application modifies the contents of that Level Zero memory allocation during the lifetime of the SYCL buffer, the behavior is undefined. The initial contents of the SYCL buffer will be the initial contents of the Level Zero memory allocation at the time of the SYCL buffer's construction.
+
+The behavior of the SYCL buffer destructor depends on the Ownership flag. As with other SYCL buffers, this behavior is triggered only when the last reference count to the buffer is dropped, as described in the core SYCL specification section 4.7.2.3, "Buffer synchronization rules".
+
+* If the ownership is keep (i.e. the application retains ownership of the Level Zero memory allocation), then the SYCL buffer destructor blocks until all work in queues on the buffer have completed. The buffer's contents is not copied back to the Level Zero memory allocation. This means that contents of the Level Zero memory allocation is not up-to-date after destruction if the SYCL buffer was used in a SYCL context which is different from the SYCL context provided to ```make_buffer``` or if the SYCL buffer was used on devices different from the device where the Level Zero memory was allocated. SYCL RT can be enforced to write data back to the original allocation by means of the ```get_native_mem()``` method of the ```interop_handle``` SYCL class which is described in the core SYCL specification section 4.10.2, "Class interop_handle" and in the Section 4.2 of this document. 

```

Comment:
It does not make sense for the destructor to block unless there is also a guarantee that the buffer contents are written back to the USM memory.  We should either:

1. Block and provide a guarantee that the buffer contents are copied back to the USM memory, or
2. Do not block and provide no guarantee about the contents of the USM memory after the buffer is destroyed.

Reply:
I though we need to block in case of "keep" ownership even if we don't copy back. User may start using device pointer (for example, write something to this memory allocation or free this allocation) as soon as buffer is destroyed.  So, if I get it right, we must be sure that all work on the buffer must be completed after buffer destruction.

Reply:
OK, that's a good point.  I agree that the destructor needs to block.

Reply:
Sorry, one more question:

> SYCL RT can be enforced to write data back to the original allocation by means of the ```get_native_mem()``` method of the ```interop_handle``` SYCL class which is described in the core SYCL specification section 4.10.2

Is this statement always true?  For example, if you called `get_native_mem` from a different device (not the device associated with the L0  memory), then this would not guarantee that the data is written to the original L0 memory allocation.  Even if you call `get_native_mem` from the same device, we do not currently guarantee that the native memory handle will be the same as the one the user passed to `make_buffer` (though I presume it would be the same in a reasonable implementation).

Do we need this sentence explaining how to cause the buffer content to be written back?

Reply:
Yes, I was trying to say the the data can be copied back if you call get_native_mem from the same device.
But after our discussion I agree that even though our current implementation is using the same pointer as the one the user passed to make_buffer, in the future it will change. So I removed the consusing statement.
Uers may refer to core sycl spec where get_native_mem behaviour is described.